### ISSUE TITLE
Handle invalid token when using only rest

### ISSUE THREAD

**Issue by natanbc** [mewna/catnip#116]
This happens when a wrong token is used 

```
jshell> catnip.rest().user().modifyCurrentUser(null, new okhttp3.OkHttpClient().newCall(new okhttp3.Request.Builder().url("https://i.warosu.org/data/jp/img/0133/96/1430233591090.jpg").build()).execute().body().bytes()).toCompletableFuture().join()
$4 ==> User (null#null)
```

**Comment by queer** [comment]
Should just validate token automatically when building catnip instances maybe?

**Comment by natanbc** [comment]
maybe check the response for invalid tokens as well

**Comment by queer** [comment]
I guess really we could just be checking here https://github.com/mewna/catnip/blob/69cc0d1072bb5da819b0e0607870ba100a86f5af/src/main/java/com/mewna/catnip/rest/RestRequester.java#L107-L112 for if it's a 401 or 403 and fail the request / log an error as needed.

**Comment by FiniteReality** [comment]
I'm a fan of validating tokens when building catnip instances. Discord.Net does this in its REST client, but IIRC it can be disabled if necessary.

Maybe another option here would be to have a 'validateToken' method which should be called on most uses of catnip, but for those who know they are only passing valid tokens (e.g. OAuth tokens generated by Discord) it is not necessary?

**Comment by queer** [comment]
Fixed in 6175a9347e9d2a8bac336d9a5e861cfadc81b863 by always validating token when building instances.
